### PR TITLE
Fix/api prevent multiple same requests

### PR DESCRIPTION
**Pull request by pandafulmanda**
Alternative to #1133, handles and prevents multiple for all requests as opposed to just those in loadable

fix is in ea53f8c, some refactoring in 35accc7, can be reset if needed.

### CODE REVIEW

File: src/helpers/api.coffee [hunk 1]
```diff
@@ -33,42 +33,100 @@ makeLocalRequest = (requestConfig) ->
...
+
+    Promise.reject(errorResponse)
+
+  handleLocalErrors: (errorResponse) ->
+    {status, statusText, config} = errorResponse
+    {method, mockMethod} = config
+
+    mockMethods = ['PUT', 'PATCH']
+
+    # Hack for local testing, fake successful PUT and PATCH
```
Comment (pandafulmanda) [inline src/helpers/api.coffee]: I kinda think it's better not to have this now that `makeRequestLocal` handles this, or this should return as a different error that makes it clear where the fake `json` should be made to handle the fake request.

File: src/helpers/api.coffee [hunk 2]
```diff
@@ -33,42 +33,100 @@ makeLocalRequest = (requestConfig) ->
...
+    if _.contains(mockMethods, mockMethod or method)
+      data = JSON.parse(errorResponse.config.data)
+      return Promise.resolve({data})
+
+    # Hack for local testing. Webserver returns 200 + HTML for 404's
```
Comment (pandafulmanda) [inline src/helpers/api.coffee]: I don't think our FE throws this error anymore?

Comment (nathanstitt) [inline src/helpers/api.coffee]: I agree - I think this is unused

**Comment by nathanstitt** [comment]
I like this better than my #1133 - it also matches what we did on CC.   If @philschatz agrees i'll close the other

**Comment by nathanstitt** [comment]
I likes, It fixes the dashboard triggering 300+ loads.  Am merging so it can be in next round of testing

**Pull request merged**